UserDetails/views.py

```diff
@@ -54,7 +51,7 @@ def post(self, request, *args, **kwargs):
         return self.render_to_response(context)
 
 
-class DiningJoinHistoryView(View, PageListView):
+class DiningJoinHistoryView(View, PageListMixin, LoginRequiredMixin):
```

Comment:
The `LoginRequiredMixin` needs to be before `View` otherwise it won't work. Also the `get` method is already decorated with `login_required`.